### PR TITLE
Migrate to reflectable

### PR DESCRIPTION
**Pull request by jakemac53**
The tests don't pass once transformed currently, but I wanted to send this out anyways to get feedback since the issues are most likely related to the reflectable transformer itself, not these changes.

Closes https://github.com/dart-lang/polymer-dart/issues/535

### CODE REVIEW

**Comment by jakemac53** [comment]
cc @eernstg

**Comment by jakemac53** [comment]
fwiw, here are the logs from pub serve https://gist.github.com/jakemac53/1b3100373f1c4df18d05

**Comment by donny-dont** [comment]
@jakemac53 does anything like https://github.com/dart-lang/smoke/issues/21 need to be added to reflectable?

**Comment by jakemac53** [comment]
@donny-dont reflectable doesn't provide a way of querying for all declarations including super class declarations at all (that I know of at least). Since I we walking up the super chain manually anyways (see `declarationsFor` in `declarations.dart`), I went ahead and implemented the same semantics there.

It is possible that we would just re-implement smoke on top of reflectable since it does provide some nice abstractions, but that is a larger endeavor ;).

File: lib/src/common/js_proxy.dart [hunk 1]
```diff
@@ -45,10 +45,12 @@ JsObject _buildJsProxy(JsProxy instance) {
   return proxy;
 }
 
-/// Query for all public fields/methods.
-final _queryOptions =
-    new smoke.QueryOptions(includeMethods: true, includeUpTo: HtmlElement,
-        excludeOverriden: true);
+class JsProxyReflectable extends Reflectable {
```
Comment (jakemac53) [inline lib/src/common/js_proxy.dart]: This is actually used by many things, and should probably be renamed, but I haven't come up with a good name yet. Suggestions welcome ;).

**Comment by donny-dont** [comment]
Okay cool I can try it out and make sure everythings working as advertised. Glad to see you back on it :+1:

**Comment by jakemac53** [comment]
@donny-dont fyi you will need to locally clone the reflectable package (https://github.com/dart-lang/reflectable) since its layout isn't currently compatible with a git dependency and its not  yet published to pub afaik. Also you will need to add the `reflectable` transformer, and the list of entry points it accepts should point to the `.dart` files, not the `.html` files like the old polymer transformer. I am also currently getting some errors in the transformer so its likely that you will have issues in pub serve/build, but a regular server should work fine.

**Comment by jakemac53** [comment]
this has been merged into the reflectable branch, closing.

**Pull request closed without merging**